scene/2d/physics_body_2d.cpp

```diff
@@ -1198,7 +1198,7 @@ Vector2 KinematicBody2D::move_and_slide(const Vector2 &p_linear_velocity, const
 						revert_motion();
 						return Vector2();
 					}
-				} else if (get_collision_normal().dot(p_floor_direction) < Math::cos(Math::deg2rad((float)45))) { //ceiling
+				} else if (get_collision_normal().dot(p_floor_direction) <= -Math::cos(Math::deg2rad(p_floor_max_angle))) { //ceiling
```

Comment:
Just a matter of style, but I think it would read nicer if you negate the floor direction (which would correspond to ceiling normal) rather than cosine.

Also, you're assuming the ceiling and ground are always parallel to each other, but this can also be relaxed by introducing a ceiling direction parameter.